### DISCUSSION TITLE
upgrading v1beta2 --> v1 | Kustomization's  prune: true ?

### DISCUSSION THREAD

**Discussion by asitha-w** [fluxcd/flux2#4596] [Q&A] [answered]
before doing the upgrade
should we set the initial boot strapped Kustomization's  prune: true ?

the reason for this question or _**worry**_ =) is  
all the existing deployments to the cluster is **held** by this **initial repo to cluster Kustomization**

so the moment new fluxcd controller commits in the new  Kustomization with the new version would there be a risk to the existing initial definition with apiVersion: [kustomize.toolkit.fluxcd.io/v1beta2] will be gone for from the repo and with that the rest of the apps

**Comment by stefanprodan** [comment, marked as answer]
The apiVersion bump is not the same as renaming the Kustomization, there is no delete & create operation. Once you upgrade Flux on the cluster, the apiVersion update happens automatically in etcd and from that moment on, your repo becomes out of sync with the cluster state until you bump the version in all YAML files.

**Reply by asitha-w** [reply]
thank you stefan for the quick response

but just to add to the notes, we are using terraform . and the terraform is planing to the below, this was the origin of this concern.

destroy the existing definition
module.fluxcd-platform.kubectl_manifest.sync["source.toolkit.fluxcd.io/v1beta2/gitrepository/flux-system/<cluster name>"] will be destroyed

and the new one will be created.
module.fluxcd-platform.kubectl_manifest.sync["source.toolkit.fluxcd.io/v1/gitrepository/flux-system/<cluster name >"] will be created

**Reply by stefanprodan** [reply]
That's really bad, you should't be using kubectl_manifest that was dropped some years ago.

**Reply by stefanprodan** [reply]
I suggest you get rid of that TF setup and start start fresh with the Flux TF provider from here https://github.com/fluxcd/terraform-provider-flux

**Reply by asitha-w** [reply]
thanks stefan. we did kind of an early adoption, trying to iron things out :)
